tests/runtime/test_service_factory.py

```diff
@@ -213,6 +216,91 @@ def test_get_and_filter_modules_respects_included_task_types_and_excluded_module
         assert "ListModule" not in str(clean_modules)
 
 
+def test_assert_compatible_does_not_raise_if_not_enabled():
+    with temp_config(
+        {
+            "runtime": {
+                "service_generation": {
+                    "backwards_compatibility": {
+                        "enabled": False,
+                    }
+                },
+            }
+        },
+        "merge",
+    ):
+        ServicePackageFactory.get_service_package(
+            ServicePackageFactory.ServiceType.INFERENCE
+        )
+
+
+def test_assert_compatible_raises_if_a_module_becomes_unsupported():
+    with tempfile.TemporaryDirectory() as workdir:
+        prev_module_file = os.path.join(workdir, "prev_modules.json")
+        random_uuid = str(uuid.uuid4())
+        with open(prev_module_file, "w", encoding="utf-8") as file:
+            json_content = {
+                "excluded_modules": {},
+                "included_modules": {
+                    "SampleTask": {
+                        random_uuid: "<class 'sample_lib.modules.sample_task.sample_implementation.PrevSampleModule'>",
+                    }
+                },
+            }
+            file.write(json.dumps(json_content, indent=4))
+        with temp_config(
+            {
+                "runtime": {
+                    "service_generation": {
+                        "backwards_compatibility": {
+                            "enabled": True,
+                            "prev_modules_path": prev_module_file,
+                        }
+                    },
+                }
+            },
+            "merge",
+        ):
+            with pytest.raises(AssertionError):
+                # this raises because PrevSampleModule will not be included in this service generation
+                ServicePackageFactory.get_service_package(
+                    ServicePackageFactory.ServiceType.INFERENCE
+                )
+
+
+def test_assert_compatible_does_not_raise_if_supported_modules_continue_to_be_supported():
+    with tempfile.TemporaryDirectory() as workdir:
+        prev_module_file = os.path.join(workdir, "prev_modules.json")
+
+        with open(prev_module_file, "w", encoding="utf-8") as file:
+            json_content = {
+                "excluded_modules": {},
```

Comment:
🤔🤔🤔🤔🤔

Do we need excluded modules in this file at all?
It looks like we don't need it for the check in this PR, and I'm not sure if there's any other check we would need it for. IIRC we had used this previously when service generation was implicit, but now that it's explicit the service generation will fail if it can't generate a handler for a module that lists a task